src/aosm/azext_aosm/common/artifact.py

```diff
@@ -2,150 +2,586 @@
+                ("Failed to push %s to %s."),
+                local_docker_image,
+                target_acr,
+            )
+            logger.debug(error, exc_info=True)
+            raise error
+        finally:
+            docker_logout_cmd = [
+                str(shutil.which("docker")),
+                "logout",
+                target_acr,
+            ]
+            self._call_subprocess_raise_output(docker_logout_cmd)
+
+    def _copy_image(
+        self,
+        config: BaseCommonParametersConfig,
+        command_context: CommandContext,
+        source_registry_login_server: str,
+        source_image: str,
+    ):
+        """
+        Copy image from one ACR to another.
+
+        Use az acr import to do the import image. Previously we used the python
+        sdk `ContainerRegistryManagementClient.registries.begin_import_image`
+        but this requires the source resource group name, which is more faff
+        at configuration time.
+
+        Neither `az acr import` or `begin_import_image` support using the username
+        and acr_token retrieved from the manifest credentials, so this uses the
+        CLI users context to access both the source registry and the target
+        Artifact Store registry. This requires either Contributor role or a
+        custom role that allows the importImage action over the whole subscription.
+
+        :param source_registry: source registry login server e.g. https://uploadacr.azurecr.io
+        :param source_image: source image including namespace and tags e.g.
+                             samples/nginx:stable
+        """
+        logger.debug("RemoteACRArtifact (copy_image) config: %s", config)
+        manifest_credentials = self._manifest_credentials(config=config, aosm_client=command_context.aosm_client)
+        # TODO (WIBNI): All oras_client is used for (I think) is to get the target_acr. Is there a simpler way to do this?
+        oras_client = self._get_oras_client(manifest_credentials=manifest_credentials)
+        target_acr = self._get_acr(oras_client)
+        try:
+            print("Copying artifact from source registry")
+            # In order to use az acr import cross subscription, we need to use a token
+            # to authenticate to the source registry. This is documented as the way to
+            # us az acr import cross-tenant, not cross-sub, but it also works
+            # cross-subscription, and meant we didn't have to make a breaking change to
+            # the format of input.json. Our usage here won't work cross-tenant since
+            # we're attempting to get the token (source) with the same context as that
+            # in which we are creating the ACR (i.e. the target tenant)
+            get_token_cmd = [str(shutil.which("az")), "account", "get-access-token"]
+            # Dont use _call_subprocess_raise_output here as we don't want to log the
+            # output
+            called_process = subprocess.run(  # noqa: S603

```

Comment:
Because we don't want to log the output (see comment on line 381).

We probably could fix up _call_subprocess_raise_output to work with this, but questionable value, and definitely not one for the refactor (the old code did this).